java/core/src/main/java/sleeper/core/statestore/AllReferencesToAFile.java

```diff
@@ -95,6 +95,15 @@ public AllReferencesToAFile withCreatedUpdateTime(Instant updateTime) {
                 .build();
     }
 
+    /**
+     * Builds an estimate of the number of records in the file by adding up all records in each file reference.
+     *
```

Comment:
I think it's worth commenting that this might be an extremely bad estimate of the number of records in a file, e.g. if we have a table with 1024 leaf partitions and one file was added with 1024 file references, and now 1023 of those have been deleted due to compactions, the sum of the number of records in each file reference is nowhere near the number of records in the file. In fact this can be such a bad estimate that I'm not sure we want it on the core API.

Reply:
There's a similar method AllReferencesToAllFiles.estimateRecordsInTable. That's also only used in tests. We can move both to the test code.

Reply:
I've moved both of those to the test code.